pkg/skaffold/initializer/build/cli.go

```diff
@@ -62,29 +63,32 @@ func (c *cliBuildInitializer) GenerateManifests() (map[GeneratedBuilderImagePair
 }
 
 func (c *cliBuildInitializer) processCliArtifacts() error {
-	pairs, err := processCliArtifacts(c.cliArtifacts)
+	pairs, workspaces, err := processCliArtifacts(c.cliArtifacts)
 	if err != nil {
 		return err
 	}
 	c.builderImagePairs = pairs
+	c.workspaces = workspaces
 	return nil
 }
 
-func processCliArtifacts(cliArtifacts []string) ([]BuilderImagePair, error) {
+func processCliArtifacts(cliArtifacts []string) ([]BuilderImagePair, []string, error) {
```

Comment:
Is there a reason to not include the workspace in the `BuilderImagePair`?

Reply:
I would prefer to do this too. The current way that I did things make it more bloated. I'll work on this change.